platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/GuestClaimsHandlerExt.java

```diff
@@ -13,129 +13,221 @@
  */
 package org.codice.ddf.ui.admin.api;
 
-import java.io.File;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.Properties;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
-import org.apache.commons.lang.StringUtils;
+import javax.annotation.Nullable;
+
+import org.codice.ddf.ui.admin.api.util.PropertiesFileReader;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import ddf.security.PropertiesLoader;
+import com.google.common.collect.ImmutableList;
 
 /**
- * GuestClaimsHandlerExt is an extension for the ConfigurationAdminMBean
- * that adds capabilities to read/handle claim and profiles for the
- * guest claims handler.
+ * GuestClaimsHandlerExt is an extension for the ConfigurationAdminMBean that adds capabilities to
+ * read/handle claim and profiles for the guest claims handler. The state in the class mirrors the
+ * loaded properties files exactly, and the data is transformed as it leaves the class to whatever
+ * format is expected.
  */
 public class GuestClaimsHandlerExt {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GuestClaimsHandlerExt.class);
+
     public static final String PROFILE_NAME_KEY = "profileName";
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(GuestClaimsHandlerExt.class);
+    public static final String AVAILABLE_PROFILES = "availableProfiles";
+
+    public static final String PROFILE_NAMES = "profileNames";
+
+    public static final String AVAILABLE_CLAIMS = "availableClaims";
+
+    public static final String IMMUTABLE_CLAIMS = "immutableClaims";
+
+    private static final String DEFAULT_NAME = "Default";
 
-    private Map<String, Object> claimsProfiles;
+    private Map<String, String> availableClaimsMap;
```

Comment:
I agree but since this work is generating a bit of tech debt, and will be re-organized after it's proven to work, I say we revisit this after the fact. All of this data lends itself better to a JSON structure instead of a bunch of flat properties files.